includes/login-register.php

```diff
@@ -159,6 +159,62 @@ function edd_log_user_in( $user_id, $user_login, $user_pass, $remember = false )
 
 }
 
+add_filter( 'wp_login_errors', 'edd_login_register_error_message', 10, 2 );
+/**
+ * Changes the WordPress login confirmation message when using EDD's reset password link.
+ *
+ * @since 2.10
+ * @param object \WP_Error $errors
+ * @param string $redirect
+ * @return void
+ */
+function edd_login_register_error_message( $errors, $redirect ) {
+	$action_is_confirm   = 'confirm' === filter_input( INPUT_GET, 'checkemail', FILTER_SANITIZE_STRING );
+	$edd_action_is_reset = 'confirm' === filter_input( INPUT_GET, 'edd_reset_password', FILTER_SANITIZE_STRING );
+	$redirect_url        = filter_input( INPUT_GET, 'edd_redirect', FILTER_SANITIZE_URL );
+	if ( $action_is_confirm && $edd_action_is_reset && $redirect_url ) {
+		$errors->remove( 'confirm' );
+		$errors->add(
+			'confirm',
+			apply_filters(
+				'edd_login_register_error_message',
+				sprintf(
+					/* translators: %s: Link to the referring page. */
+					__( 'Follow the instructions in the confirmation email you just received, then <a href="%s">return to what you were doing</a>.', 'easy-digital-downloads' ),
+					esc_url( $redirect_url )
+				),
+				$redirect_url
+			),
+			'message'
+		);
+	}
+
+	return $errors;
+}
+
+/**
+ * Gets the lost password URL, customized for EDD. Using this allows the password
+ * reset form to redirect to the login screen with the EDD custom confirmation message.
+ *
+ * @since 2.10
+ * @return string
+ */
+function edd_get_lostpassword_url() {
+	$url = wp_get_raw_referer();
```

Comment:
This function was actually introduced in WP 4.5, so we need something as a fallback for 4.4, since we do still support that. @ashleyfae is it okay if I copy over the [logic from the function](https://developer.wordpress.org/reference/functions/wp_get_raw_referer/)?

Reply:
Looking at this some more..

`wp_get_raw_referer()` seems to differ from `wp_get_referer()` for two reasons:

1. `wp_get_raw_referer()` doesn't care if it's the same as the current URL.
2. `wp_get_raw_referer()` does not validate the redirect (using `wp_validate_redirect()` ).

I suspect those are the two reasons why they say:

```
Do not use for redirects, use wp_get_referer() instead.
```

I'm not sure if that's something we should be ignoring just because it was a convenient "fix" for `wp_get_referer()` returning `false`. Maybe we should look more into why that was?

Reply:
I think part of the issue is how and where we are getting and using the redirect URL. I think generally, the redirect retrieval and output are happening in one place, so it makes sense that the referer URL being the same would be problematic. In our situation, however, we are getting the URL in one place, but sending it to a second page (actually onto a third) to be used. So I think it makes sense why the one is not working for our situation and the other is.

Maybe we should also incorporate the `wp_validate_redirect` function before using the URL, though, so there is further checking going on.